akka-remote/src/main/scala/akka/remote/artery/compress/HeavyHitters.scala

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright (C) 2009-2016 Lightbend Inc. <http://www.lightbend.com>
+ */
+
+package akka.remote.artery.compress
+
+import scala.collection.immutable
+
+// TODO describe / think about how we could handle an actor dying and removing from "top"
+// TODO this is the scenario when "super traffic actor" goes away and we want to not use up the slot in HH
+
+/**
+ * INTERNAL API
+ *
+ * Mutable.
+ *
+ * Keeps a number of specific heavy hitters around in memory.
+ *
+ * See also Section 5.2 of http://dimacs.rutgers.edu/~graham/pubs/papers/cm-full.pdf
+ * for a discussion about the assumptions made and guarantees about the Heavy Hitters made in this model.
+ * We assume the Cash Register model in which there are only additions, which simplifies HH detecion significantly.
+ */
+private[remote] sealed class HeavyHitters[T](strategy: HeavyHittersStrategy[T]) {
+  // TODO technically could be optimised depending on strategy (priority queue?) (store in order of counts, drop last if exceeded size)
+  private[this] var _items = List.empty[HeavyHitter[T]]
```

Comment:
I'll move to the PQ for now, removed all the other strategies along the way, will measure